src/test/java/guitests/UseGlobalConfigsTest.java

```diff
@@ -85,10 +86,9 @@ public void globalConfigTest() {
         assertEquals(2, lastOpenFilters.size());
         assertEquals("", lastOpenFilters.get(0));
         assertEquals("repo:dummy2/dummy2", lastOpenFilters.get(1));
-        // Last viewed repositories
-        List<String> lastViewedRepositories = testPref.getLastViewedRepositories();
-        assertEquals("dummy/dummy", lastViewedRepositories.get(0));
-        assertEquals("dummy2/dummy2", lastViewedRepositories.get(1));
+        // Last viewed repository
+        RepositoryId lastViewedRepository = testPref.getLastViewedRepository().get();
+        assertEquals("dummy/dummy", lastViewedRepository.generateId());
```

Comment:
Are we testing for all possible cases? There is a lot of functional code and very little test code in this PR.